homeassistant/components/tuya/vacuum.py

```diff
@@ -112,18 +112,13 @@ def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> Non
             self._supported_features |= SUPPORT_FAN_SPEED
             self._fan_speed_type = EnumTypeData.from_json(function.values)
 
-        if function := device.function.get(DPCode.ELECTRICITY_LEFT):
+        if DPCode.ELECTRICITY_LEFT in self.device.status:
             self._supported_features |= SUPPORT_BATTERY
-            self._battery_level_type = IntegerTypeData.from_json(function.values)
 
     @property
     def battery_level(self) -> int | None:
         """Return Tuya device state."""
-        if self._battery_level_type is None or not (
-            status := self.device.status.get(DPCode.ELECTRICITY_LEFT)
-        ):
-            return None
-        return round(self._battery_level_type.scale_value(status))
+        return self.device.status.get(DPCode.ELECTRICITY_LEFT)
```

Comment:
This change is incorrect, as the value is no longer scaled against its data type.

Reply:
@frenck, happy New Year!

According to Tuya Documents, the [Instruction Set](https://developer.tuya.com/en/docs/iot/fsd?id=K9gf487ck1tlo) does not include battery information. Being only included in the [Status Set](https://developer.tuya.com/en/docs/iot/ssd?id=K9gf48qts32n0).

The ELECTRICITY_LEFT is in `status_range`, `device.function.get()` doesn't return this information.

So I changed from `device.function.get()` to `device.status_range.get()` as available in the [SDK](https://github.com/tuya/tuya-iot-python-sdk/blob/2a3e473446c2a913d4336054431dbf0540c40917/tuya_iot/device.py#L104) to get the battery info.

This solves the problem.
![image](https://user-images.githubusercontent.com/31328123/147864989-3698e21f-3e4a-40ce-a994-61c498aa1d02.png) 
![image](https://user-images.githubusercontent.com/31328123/147864997-7c241691-18b7-413d-b908-5134a8fd233c.png)


Here is the device information returned by the API:
<details><summary>Instruction Set</summary>

````
{
  "result": {
    "category": "sd",
    "functions": [
      {
        "code": "power_go",
        "desc": "{}",
        "name": "清扫开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "mode",
        "desc": "{\"range\":[\"smart\",\"pose\",\"zone\",\"part\",\"chargego\",\"wall_follow\",\"pick_zone\"]}",
        "name": "工作模式",
        "type": "Enum",
        "values": "{\"range\":[\"smart\",\"pose\",\"zone\",\"part\",\"chargego\",\"wall_follow\",\"pick_zone\"]}"
      },
      {
        "code": "direction_control",
        "desc": "{\"range\":[\"forward\",\"backward\",\"turn_left\",\"turn_right\",\"stop\"]}",
        "name": "清扫方向",
        "type": "Enum",
        "values": "{\"range\":[\"forward\",\"backward\",\"turn_left\",\"turn_right\",\"stop\"]}"
      },
      {
        "code": "seek",
        "desc": "{}",
        "name": "寻找机器",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "suction",
        "desc": "{\"range\":[\"strong\",\"normal\",\"quiet\"]}",
        "name": "吸力选择",
        "type": "Enum",
        "values": "{\"range\":[\"strong\",\"normal\",\"quiet\"]}"
      },
      {
        "code": "pause",
        "desc": "{}",
        "name": "暂停开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "reset_map",
        "desc": "{}",
        "name": "重置地图",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "switch_disturb",
        "desc": "{}",
        "name": "勿扰开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "volume_set",
        "desc": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}",
        "name": "音量设置",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      }
    ],
    "status": [
      {
        "code": "power_go",
        "name": "清扫开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "mode",
        "name": "工作模式",
        "type": "Enum",
        "values": "{\"range\":[\"smart\",\"pose\",\"zone\",\"part\",\"chargego\",\"wall_follow\",\"pick_zone\"]}"
      },
      {
        "code": "direction_control",
        "name": "清扫方向",
        "type": "Enum",
        "values": "{\"range\":[\"forward\",\"backward\",\"turn_left\",\"turn_right\",\"stop\"]}"
      },
      {
        "code": "status",
        "name": "当前状态",
        "type": "Enum",
        "values": "{\"range\":[\"standby\",\"smart_clean\",\"zone_clean\",\"part_clean\",\"paused\",\"goto_pos\",\"wall_clean\",\"goto_charge\",\"charging\",\"charge_done\",\"cleaning\",\"pick_zone_clean\",\"pos_arrived\",\"pos_unarrive\",\"sleep\"]}"
      },
      {
        "code": "electricity_left",
        "name": "剩余电量",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      },
      {
        "code": "seek",
        "name": "寻找机器",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "suction",
        "name": "吸力选择",
        "type": "Enum",
        "values": "{\"range\":[\"strong\",\"normal\",\"quiet\"]}"
      },
      {
        "code": "clean_area",
        "name": "清扫面积",
        "type": "Integer",
        "values": "{\"unit\":\"㎡\",\"min\":0,\"max\":9999,\"scale\":0,\"step\":1}"
      },
      {
        "code": "clean_time",
        "name": "清扫时间",
        "type": "Integer",
        "values": "{\"unit\":\"min\",\"min\":0,\"max\":9999,\"scale\":0,\"step\":1}"
      },
      {
        "code": "pause",
        "name": "暂停开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "reset_map",
        "name": "重置地图",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "switch_disturb",
        "name": "勿扰开关",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "volume_set",
        "name": "音量设置",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      }
    ]
  },
  "success": true,
  "t": 1641089070612
}
````
</details>

<details><summary>Status Set</summary>

````
{
  "result": [
    {
      "code": "power_go",
      "value": false
    },
    {
      "code": "mode",
      "value": "smart"
    },
    {
      "code": "direction_control",
      "value": "foward"
    },
    {
      "code": "status",
      "value": "sleep"
    },
    {
      "code": "electricity_left",
      "value": 99
    },
    {
      "code": "seek",
      "value": true
    },
    {
      "code": "suction",
      "value": "normal"
    },
    {
      "code": "clean_area",
      "value": 0
    },
    {
      "code": "clean_time",
      "value": 0
    },
    {
      "code": "pause",
      "value": false
    },
    {
      "code": "reset_map",
      "value": true
    },
    {
      "code": "switch_disturb",
      "value": false
    },
    {
      "code": "volume_set",
      "value": 9
    }
  ],
  "success": true,
  "t": 1641089947240
}
````
</details>

Please review.

Thanks!